qiime/workflow/preprocess.py

```diff
@@ -121,14 +124,27 @@ def create_commands_slf(all_files, demultiplexing_method, output_dir,
         output directory names.
     sampleid_indicator: Split on this character in input fastq filenames to
         generate output SampleID name.
+
+    Raises
+    ------
+    IOError
+        If the output_dir doesn't exist.
     """
 
+    # we need the folder to exist so we can write the seqs.txt, barcodes.txt,
+    # sample_ids.txt and maps.txt files
+    if not exists(output_dir):
+        raise IOError("%s directory doesn't exist" % output_dir)
+
     commands = []
     read_files = []
     barcode_files = []
     mapping_files = []
     sample_ids = []
 
+    params_dict = {}
+    path_builder = lambda x: join(output_dir, x)
```

Comment:
Hmmm, I don't think I can do that, if I were to do that then output_dir
would be appended to the name of my file i.e. path_builder('foo.hdf5')
would result in 'foo.hdf5/output_dir_name'.

On (Aug-28-15|16:20), adamrp wrote:
> >      commands = []
> >      read_files = []
> >      barcode_files = []
> >      mapping_files = []
> >      sample_ids = []
> >  
> > +    params_dict = {}
> > +    path_builder = lambda x: join(output_dir, x)
> 
> `functools.partial`?
> ```python
> path_builder = partial(join, output_dir)
> ```
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiime/pull/2077/files#r38253638